pandas/core/index.py

```diff
@@ -2580,7 +2580,10 @@ def drop(self, labels, errors='raise'):
     @Appender(_shared_docs['drop_duplicates'] % _index_doc_kwargs)
     def drop_duplicates(self, take_last=False):
         result = super(Index, self).drop_duplicates(take_last=take_last)
-        return self._constructor(result)
+        if self.name is not None:
+            return self._constructor(result, name=self.name)
+        else:
+            return self._constructor(result, names=self.names)
```

Comment:
though looking at this now, I think ``MultiIndex._shallow_copy`` is wrong

let me see if I can fix this